From 5bd7b5d04435bd593349825973ce32290f5f604d Mon Sep 17 00:00:00 2001
From: Changqing Li <changqing.li@windriver.com>
Date: Wed, 25 Jul 2018 09:55:25 +0800
Subject: [PATCH] samba: cmocka.h: fix musl libc conflicting types error

Fix build on qemumips64(el)

taken from:
[PATCH] libldb: fix musl libc conflicting types error

/third_party/cmocka/cmocka.h:126:28: error: conflicting types for 'uintptr_t'
            typedef unsigned int uintptr_t;
                                 ^~~~~~~~~
use __DEFINED_uintptr_t in alltypes.h to check if uintptr already defined

Upstream-Status: Pending

Signed-off-by: Changqing Li <changqing.li@windriver.com>
Signed-off-by: Andrea Adami <andrea.adami@gmail.com>

Taken from [PATCH] samba: cmocka.h: fix musl libc conflicting types error
https://git.openembedded.org/meta-openembedded/commit/meta-networking/recipes-connectivity/samba?id=b758150d30a5fbdc47c465d0e9dca3a7e6d71f3c

Signed-off-by: Armin Kuster <akuster808@gmail.com>

This fix is not in cmocka, so pending pending??


Index: git/include/cmocka.h
===================================================================
--- git.orig/include/cmocka.h
+++ git/include/cmocka.h
@@ -116,7 +116,7 @@ typedef uintmax_t LargestIntegralType;
     ((LargestIntegralType)(value))
 
 /* Smallest integral type capable of holding a pointer. */
-#if !defined(_UINTPTR_T) && !defined(_UINTPTR_T_DEFINED)
+#if !defined(__DEFINED_uintptr_t)
 # if defined(_WIN32)
     /* WIN32 is an ILP32 platform */
     typedef unsigned int uintptr_t;
@@ -140,9 +140,8 @@ typedef uintmax_t LargestIntegralType;
 #  endif /* __WORDSIZE */
 # endif /* _WIN32 */
 
-# define _UINTPTR_T
-# define _UINTPTR_T_DEFINED
-#endif /* !defined(_UINTPTR_T) || !defined(_UINTPTR_T_DEFINED) */
+# define __DEFINED_uintptr_t
+#endif /* !defined(__DEFINED_uintptr_t) */
 
 /* Perform an unsigned cast to uintptr_t. */
 #define cast_to_pointer_integral_type(value) \
